File: java/src/test/java/supermarket/CashDeskSpecsDriver.java
package supermarket;

import java.util.List;

import supermarket.model.checkout.CashRegister;
import supermarket.model.checkout.ICashRegister;
import supermarket.model.checkout.Receipt;
import supermarket.model.core.EAN;
import supermarket.model.core.Product;

public class CashDeskSpecsDriver {
	private IProductRepository repository;
	private ICashRegister cashRegister;
	private Receipt receipt;

	public CashDeskSpecsDriver(ProductRepository repository, CashRegister cashRegister) {
		this.repository = repository;
		this.cashRegister = cashRegister;
	}

	public Receipt getReceipt() {
		return receipt;
	}

	public void setReceipt(Receipt receipt) {
		this.receipt = receipt;
	}

	public void addProducts(List<Product> products) {
		repository.addAll(products);
	}

	public void scanProductsByName(List<String> productNames) {
		for (String name : productNames) {
			Product product = repository.getProductByName(name);
			EAN ean = EANGenerator.getEANFromHash(product);

			// TODO: Add code to scan the product with the cash register
			// cashRegister...
		}
	}

	public Receipt checkout() {
		// TODO: Add code to get the receipt from the cash register
		// return cashRegister...
		throw new RuntimeException("Not yet implemented now");
	}

}

File: CSharp/Supermarket.Specs/Bindings/CashDeskSpecsDriver.cs
using System;
using System.Collections.Generic;
using Supermarket.Model.Checkout;
using Supermarket.Model.Core;

namespace Supermarket.Specs.Bindings
{
    public class CashDeskSpecsDriver
    {
        private readonly IProductRepository _repository;
        private readonly ICashRegister _cashRegister;

        public CashDeskSpecsDriver(IProductRepository repository, ICashRegister cashRegister)
        {
            _repository = repository;
            _cashRegister = cashRegister;
        }

        public void AddProducts(IEnumerable<Product> products)
        {
            _repository.AddRange(products);
        }

        public void ScanProductsByName(IEnumerable<string> productNames)
        {
            foreach (var productName in productNames)
            {
                var product = _repository.GetProductByName(productName);
                var ean = product.Number;

                // TODO: Add code to scan the product with the cash register
                // _cashRegister...
            }
        }

        public Receipt Receipt { get; set; }

        public Receipt Checkout()
        {
            // TODO: Add code to get the receipt from the cash register
            // return _cashRegister...
            throw new NotImplementedException();
        }
    }
}